sabelotodo/tests/test_routes.py

```diff
@@ -1,174 +1,247 @@
     specified ID. """
 
-    items = [Item(name="a", order=0, done=False),
-             Item(name="b", order=2, done=True),
-             Item(name="c", order=1, done=False)]
-
+    items = items_schema.load(VALID_ITEM_DATA)
     _db.session.add_all(items)
     _db.session.commit()
 
     selection = items[idx]
     remainder = items[:idx] + items[idx+1:]
     return_value = test_client.delete('/item/%s' % selection.id)
+    returned_item = item_schema.loads(return_value.data)
 
-    # The request succeeds.
-    assert return_value.status_code == 200
+    assert return_value.status_code == 200, "The request succeeds."
+    assert returned_item == selection, "We get back the item we expect."
+    assert Item.query.all() == remainder, \
+        "The remaining items are the ones we expect."
 
-    # The items left in the database are the ones we expect.
-    assert sorted(Item.query.all()) == sorted(remainder)
-
-    # There is no longer an item in the database with the ID of the one we
-    # selected for deletion.
     get_attempt = test_client.get('/item/%s' % selection.id)
-    assert get_attempt.status_code == 404
-
-
-@pytest.mark.parametrize("source_dict",
-                         [{'name': 'a',
-                           'order': 57,
-                           'done': True},
-                          {'name': 'a',
-                           'order': 1,
-                           'done': False,
-                           'description': 'Lorem ipsum dolor sit amet',
-                           'start_date': datetime.datetime(2020, 1, 1, 0, 0),
-                           'due_date': datetime.datetime(2020, 3, 1, 0, 0),
-                           'end_date': datetime.datetime(2020, 5, 1, 0, 0)}
-                          ])
-def test_post_item_route_with_valid_input(test_client, _db, source_dict):
+    assert get_attempt.status_code == 404, \
+        "Attempting to get the item we deleted by ID fails."
+
+
+@pytest.mark.parametrize("idx, overwrite_dict",
+                         product(range(len(VALID_ITEM_DATA)),
+                                 VALID_OVERWRITE_DATA))
+def test_patch_itemid_route_with_valid_id(test_client, _db, idx, overwrite_dict):
+    """ A PATCH request to /item/<id> overwrites the specified
+    fields of the item with the specified ID. """
+
+    items = items_schema.load(VALID_ITEM_DATA)
+    _db.session.add_all(items)
+    _db.session.commit()
+
+    selection = items[idx]
+    return_value = test_client.patch('/item/%s' % selection.id, json=overwrite_dict)
+
+    assert return_value.status_code == 200, "The request succeeds."
+
+    expected_dict = {**item_schema.dump(selection), **overwrite_dict}
+    expected_item = item_schema.load(expected_dict)
+    assert selection == expected_item, \
+        "The selected item changes in the expected way."
+
+
+@pytest.mark.parametrize("idx, overwrite_dict",
+                         product(range(len(VALID_ITEM_DATA)), INVALID_OVERWRITE_DATA))
```

Comment:
I would put a comment where you define VALID_ITEM_DATA specifying that if you change it, you must also add something to INVALID_OVERWRITE_DATA. Otherwise I think this test will start failing.

Reply:
Nope, it still works if you change one or the other. The test doesn't depend on them being the same length. It independently picks a valid item to overwrite and an invalid value to overwrite it with.